### PR TITLE
feat: integrate RS Drive and GroveDB

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

To provide cryptographically verifiable proofs for document secondary indices and atomicity for state commit we need to migrate from MongoDB to a homemade DB made specifically for this purpose - [GroveDB](https://github.com/dashevo/grovedb)

## What was done?

- https://github.com/dashevo/platform/pull/227
- removed double tree workaround
- removed MongoDB
- integrated RS Drive for Document and DataContract repositories
- introduced signed state and signed state rotation
- disabled proofs
- removed multi-tree proofs
- removed root tree (GroveDB has it)
- stored all data in single instance of GroveDB and atomically commit it
- refactored init chain handler
- introduced query constraints
- introduced block execution context stack
- and much more

## How Has This Been Tested?

Running local tests

## Breaking Changes

- New state is not compatible with previous versions
- Document queries have limitations compared with previous versions
- Proofs are temporary disabled 

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/dashmate/configs/migrations.js [hunk 1]
```diff
@@ -297,6 +297,7 @@ module.exports = {
     return configFile;
   },
   '0.22.0': (configFile) => {
+    // TODO: Handle the changes in schema and configs
```
Comment (shuplenkov) [inline packages/dashmate/configs/migrations.js]: is it still actual?

File: packages/js-drive/lib/storage/GroveDBStore.js [hunk 2]
```diff
@@ -0,0 +1,367 @@
+// const pino = require('pino');
+const { createHash } = require('crypto');
+
+const logger = require('../util/noopLogger');
+//
+// const logger = pino({
```
Comment (shuplenkov) [inline packages/js-drive/lib/storage/GroveDBStore.js]: Do we need the second logger?

File: packages/js-drive/scripts/abci.js [hunk 3]
```diff
@@ -25,16 +22,6 @@ const banner = '\n ____       ______      ____        __  __                 ___
 console.log(chalk.hex('#008de4')(banner));
```
Comment (shuplenkov) [inline packages/js-drive/scripts/abci.js]: Do we need console.log here?

Comment (jawid-h) [reply packages/js-drive/scripts/abci.js:22]: Yep, it's banner

File: packages/dapi/lib/externalApis/drive/DriveClient.js [hunk 4]
```diff
@@ -100,8 +100,8 @@ class DriveClient {
    * @param {Object} options.where - Mongo-like query
    * @param {Object} options.orderBy - Mongo-like sort field
    * @param {number} options.limit - how many objects to fetch
-   * @param {number} options.startAt - number of objects to skip
-   * @param {number} options.startAfter - exclusive skip
+   * @param {Buffer} options.startAt - number of objects to skip
```
Comment (shumkov) [inline packages/dapi/lib/externalApis/drive/DriveClient.js:103]: JSDoc

File: packages/dashmate/configs/migrations.js (hunk 1 above)
Comment (shumkov) [reply packages/dashmate/configs/migrations.js:300]: Yes

File: packages/js-drive/lib/blockExecution/BlockExecutionContextStackRepository.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      blockExecutionContextStack.setContexts(blockExecutionContexts);
+
+      return blockExecutionContextStack;
+    } catch (e) {
+      if (e.type === 'NotFoundError') {
+        return new BlockExecutionContextStack();
+      }
+
+      throw e;
+    }
```
Comment (shumkov) [inline packages/js-drive/lib/blockExecution/BlockExecutionContextStackRepository.js]: We don't need to check for this anymore.

File: packages/js-drive/test/unit/abci/handlers/beginBlockHandlerFactory.spec.js [hunk 6]
```diff
@@ -152,19 +154,30 @@ describe('beginBlockHandlerFactory', () => {
     }
   });
 
-  it('should abort already started transactions', async () => {
-    blockExecutionDBTransactionsMock.isStarted.returns(true);
+  it('should abort already started transactions', async function it() {
```
Comment (shumkov) [inline packages/js-drive/test/unit/abci/handlers/beginBlockHandlerFactory.spec.js]: Don't we rollback them instead of aborting?

**Review by jawid-h — APPROVED** [review]
Huge pr! :+1:

**Pull request merged**